bin/rule_parser.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
```

Comment:
I think we should add the license header and copyright info to the headers of files. Can you check which copyright is correct?